tests/gtest_retry.cpp

```diff
@@ -0,0 +1,74 @@
+#include <gtest/gtest.h>
+#include "behaviortree_cpp_v3/bt_factory.h"
+#include "behaviortree_cpp_v3/decorators/retry_node.h"
+
+using BT::NodeStatus;
+
+class FinishAfterN : public BT::SyncActionNode
+{
+public:
+  FinishAfterN(const std::string& name, int N):
+    BT::SyncActionNode(name, {}), counter_(N)
+  {
+  }
+
+  NodeStatus tick() override
+  {
+    if (counter_-- > 0)
+    {
+      std::string msg = return_value == NodeStatus::SUCCESS ? "SUCCESS" : "FAILURE";
+      std::cout << name() << "; " << msg << std::endl;
+      return return_value;
+    }
+    else
+    {
+      if (return_value == NodeStatus::SUCCESS)
+      {
+        std::cout << name() << "; FAILURE" << std::endl;
+        return NodeStatus::FAILURE;
+      }
+      else
+      {
+        std::cout << name() << "; SUCCESS" << std::endl;
+        return NodeStatus::SUCCESS;
+      }
+    }
+  }
```

Comment:
R: can simplify to sth like
```suggestion
    // flip returned value after calling tick() N times  
    const NodeStatus return_val = counter_-- > 0 ? return_value : (return_value == NodeStatus::SUCCESS ? NodeStatus::FAILURE : NodeStatus::SUCCESS);
    const std::string msg = return_val == NodeStatus::SUCCESS ? "SUCCESS" : "FAILURE";
    std::cout << name() << "; " << msg << std::endl;
    return return_val;
```
maybe renaming `return_value` to sth like `initial_return_value` would also make it more clear what's happening